### PR TITLE
Verify is holiday date / presave in calendar

### PR DESCRIPTION
**Pull request by adudekyeti**
Verify is holiday date / presave in calendar

### CODE REVIEW

File: public_html/layouts/basic/modules/Calendar/resources/Edit.js [hunk 1]
```diff
@@ -492,6 +492,56 @@ Vtiger_Edit_Js("Calendar_Edit_Js", {}, {
...
+				let messageContent = app.vtranslate("JS_DATES_SELECTED_INPUTS") + '"' + holidaysArray.join(', ') + '"' + app.vtranslate("JS_DATES_SELECTED_HOLIDAYS");
+				Vtiger_Helper_Js.showConfirmationBox({'message': messageContent}).done(function () {
+					lockSave = false;
+					form.submit();
+				});
+			} else {
+				form.submit();
+			}
+		});
+		lockSave = true;
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Calendar/resources/Edit.js]: ??

File: modules/Vtiger/actions/Fields.php [hunk 2]
```diff
@@ -211,4 +212,36 @@ public function findAddress(\App\Request $request)
...
+	 */
+	public function verifyIsHolidayDate(\App\Request $request)
+	{
+		if ($this->fieldModel->getFieldDataType() === 'datetime' || $this->fieldModel->getFieldDataType() === 'date') {
+			$response = new Vtiger_Response();
+			$data = ['isHolidayDate' => false];
+			if ($request->isEmpty('date', true)) {
+				$data['message'] = \App\Language::translate('LBL_NO_DATE');
+			}
+			if (empty($data['message'])) {
```
Comment (rskrzypczak) [inline modules/Vtiger/actions/Fields.php]: else

File: public_html/layouts/basic/modules/Calendar/resources/Edit.js [hunk 3]
```diff
@@ -492,6 +492,56 @@ Vtiger_Edit_Js("Calendar_Edit_Js", {}, {
...
 	},
+	/**
+	 * Function validate is holiday day
+	 * @param {jQuery} form
+	 * @returns {Array}
+	 */
+	validateHolidayDate(form) {
+		let fields = form.find('[name="date_start"], [name="due_date"]'),
+			fieldHoliday = [];
+		$.each(fields, function (index, fieldObj) {
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Calendar/resources/Edit.js]: No each. One request

File: modules/Vtiger/actions/Fields.php [hunk 4]
```diff
@@ -211,4 +212,36 @@ public function findAddress(\App\Request $request)
...
+
+	/**
+	 * Verify is holiday date.
+	 *
+	 * @param \App\Request $request
+	 *
+	 * @throws \App\Exceptions\NoPermitted
+	 */
+	public function verifyIsHolidayDate(\App\Request $request)
+	{
```
Comment (rskrzypczak) [inline modules/Vtiger/actions/Fields.php]: Verify between dates

File: public_html/layouts/basic/modules/Calendar/resources/Edit.js (hunk 3 above)
Comment (adudekyeti) [reply public_html/layouts/basic/modules/Calendar/resources/Edit.js]: Done

File: modules/Vtiger/actions/Fields.php (hunk 4 above)
Comment (adudekyeti) [reply modules/Vtiger/actions/Fields.php]: Done

File: public_html/layouts/basic/modules/Calendar/resources/Edit.js [hunk 5]
```diff
@@ -492,6 +492,54 @@ Vtiger_Edit_Js("Calendar_Edit_Js", {}, {
...
+	},
+	/**
+	 * Register pre save event
+	 * @param {jQuery} form
+	 */
+	registerRecordPreSaveEventEvent: function (form) {
+		const self = this;
+		let lockSave = true;
+		form.on(Vtiger_Edit_Js.recordPreSave, function (e, data) {
+			let holidaysArray = self.validateHolidayDate(form);
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Calendar/resources/Edit.js]: Deffered

File: modules/Vtiger/actions/Fields.php [hunk 6]
```diff
@@ -211,4 +212,35 @@ public function findAddress(\App\Request $request)
...
+						$data = ['isHolidayDate' => true];
+					}
+				} catch (\App\Exceptions\FieldException $e) {
+					$data = ['isHolidayDate' => false];
+				}
+			}
+			$response->setResult($data);
+			$response->emit();
+		} else {
+			throw new \App\Exceptions\NoPermitted('LBL_NO_PERMISSIONS_TO_FIELD');
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/actions/Fields.php]: ERR_....

File: public_html/layouts/basic/modules/Calendar/resources/Edit.js [hunk 7]
```diff
@@ -492,6 +492,57 @@ Vtiger_Edit_Js("Calendar_Edit_Js", {}, {
...
+	 * @param {jQuery} form
+	 */
+	registerRecordPreSaveEventEvent: function (form) {
+		const self = this;
+		let lockSave = true;
+		form.on(Vtiger_Edit_Js.recordPreSave, function (e, data) {
+			self.validateHolidayDate(form).done(function (isHoliday) {
+				if (lockSave && isHoliday) {
+					e.preventDefault();
+					Vtiger_Helper_Js.showConfirmationBox({'message': app.vtranslate("JS_DATES_SELECTED_HOLIDAYS")}).done(function () {
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Calendar/resources/Edit.js]: Vtiger_Helper_Js.showConfirmationBox >> app.showAlert

**Pull request merged**